tests/integ/utils/sql_counter.py

```diff
@@ -395,15 +384,29 @@ def sql_count_checker(
         )
     }
 
-    result = func(*args, **kwargs)
-    try:
-        sql_counter.expects(**count_kwargs)
-    finally:
-        try:
-            sql_counter.close()
-        except Exception:
-            pass
-    return result
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            sql_counter = SqlCounter(
+                no_check=no_check,
```

Comment:
It looks like all tests that contain 'fallback_count=#' are disabled from being counted by pytest.mark.xfail:

@pytest.mark.xfail(
    reason="SNOW-1336091: Snowpark pandas cannot run in sprocs until modin 0.28.1 is available in conda",
    strict=True,
    raises=RuntimeError,
)

but when you actually run the tests, they fail with some other error such as NotImplementedError.

I believe a long time ago we would go through a 'fallback' to run the code using pandas within a sproc.  However, it looks like this was removed by https://github.com/snowflakedb/snowpark-python/pull/1542/files .  I'd recommend to check with @sfc-gh-nkrishna here.  Perhaps the disabled tests need to be re-evaluated to validate not implemented error rather than skipped in this way.